snapdtool/cmdutil_test.go

```diff
@@ -104,6 +128,31 @@ func (s *cmdutilSuite) TestCommandFromSystemSnapNotrick(c *C) {
 	})
 }
 
+func (s *cmdutilSuite) TestCommandFromSystemSnapLDsoTrickNeeded(c *C) {
+	defer release.MockReleaseInfo(&release.OS{ID: "fedora"})()
+	dirs.SetRootDir(dirs.GlobalRootDir)
+
+	defer snapdtool.MockElfInterp(func(p string) (string, error) {
+		c.Assert(p, Equals, filepath.Join(dirs.SnapMountDir, "snapd/current/usr/bin/xdelta3"))
+		return filepath.Join(dirs.GlobalRootDir, "/snap/snapd/current/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2"), nil
```

Comment:
Can this cause issues on non-x64 archs? I don't know if we run unit tests as a part of the build on LP